src/model_io/urdf/src/URDFModelExport.cpp

```diff
@@ -0,0 +1,295 @@
+                  <<  typeid(joint).name() << " to a URDF joint " << std::endl;
+        return false;
+    }
+
+    // origin
+    exportTransform(joint->getRestTransform(parentLink->getIndex(), childLink->getIndex()), joint_xml);
+
+    if (joint->getNrOfDOFs() != 0)
+    {
+        Axis axis;
+
+        // Check that the axis of the joint is supported by URDF
+        if (dynamic_cast<const RevoluteJoint*>(joint))
+        {
+            const RevoluteJoint* revJoint = dynamic_cast<const RevoluteJoint*>(joint);
+            axis = revJoint->getAxis(childLink->getIndex());
+        }
+        else if (dynamic_cast<const PrismaticJoint*>(joint))
+        {
+            const PrismaticJoint* prismJoint = dynamic_cast<const PrismaticJoint*>(joint);
+            axis = prismJoint->getAxis(childLink->getIndex());
+        }
+        else
+        {
+            std::cerr << "[ERROR] URDFModelExport: Impossible to convert joint of type "
+                      <<  typeid(joint).name() << " to a URDF joint " << std::endl;
+            return false;
+        }
+
+        // Check that the axis is URDF-compatible
+        if (toEigen(axis.getOrigin()).norm() > 1e-7)
+        {
+            std::cerr << "[ERROR] URDFModelExport: Impossible to convert joint "
+                      <<   model.getJointName(joint->getIndex()) << " to a URDF joint without moving the link frame of link "
+                      << model.getLinkName(childLink->getIndex()) << " , because its origin is "
+                      << axis.getOrigin().toString()  << std::endl;
+            return false;
+        }
+
+        // axis
+        xmlNodePtr axis_xml = xmlNewChild(joint_xml, NULL, BAD_CAST "axis", NULL);
+        std::string bufStr;
+        ok = ok && vectorToString(axis.getDirection(), bufStr);
+        xmlNewProp(axis_xml, BAD_CAST "xyz", BAD_CAST bufStr.c_str());
+    }
+
+    // parent
+    xmlNodePtr parent_xml = xmlNewChild(joint_xml, NULL, BAD_CAST "parent", NULL);
+    xmlNewProp(parent_xml, BAD_CAST "link", BAD_CAST model.getLinkName(parentLink->getIndex()).c_str());
+
+    // child
+    xmlNodePtr child_xml = xmlNewChild(joint_xml, NULL, BAD_CAST "child", NULL);
+    xmlNewProp(child_xml, BAD_CAST "link", BAD_CAST model.getLinkName(childLink->getIndex()).c_str());
+
+    // TODO(traversaro) : handle joint limits and friction
+
+    return ok;
+}
+
+bool URDFStringFromModel(const iDynTree::Model & model,
+                         std::string & urdf_string,
+                         const ModelExporterOptions options)
+{
+    bool ok = true;
+
+    xmlDocPtr urdf_xml = xmlNewDoc(BAD_CAST "1.0");
+    xmlNodePtr robot = xmlNewNode(NULL, BAD_CAST "robot");
+    // TODO(traversaro) properly export this :)
+    xmlNewProp(robot, BAD_CAST "name", BAD_CAST "iDynTreeURDFModelExportModelName");
+    xmlDocSetRootElement(urdf_xml, robot);
+
+    // TODO(traversaro) : We are assuming that the model has no loops,
```

Comment:
This must be mentioned in the documentation, explicitly. Since I notice there are users of iDynTree in the cable robots or parallel mechanisms community as well.

Reply:
I think this should be in the main readme of idyntree not only in this pr
I thought it was there but I could not find a mention to it now.

Reply:
Related issue: https://github.com/robotology/idyntree/issues/555 .